src/test/compile-fail/const-eval-span.rs

```diff
@@ -14,7 +14,7 @@
 struct S(i32);
 
 const CONSTANT: S = S(0);
-//~^ ERROR: constant evaluation error: non-constant path in constant expression [E0080]
+//~^ ERROR: constant evaluation error: call on struct [E0080]
```

Comment:
clarification: What I mean by that is: Is this a deliberate design decision to make this illegal?  Or is it just an artifact of the current rustc implementation?